racetrack_commons/racetrack_commons/plugin/loader.py

```diff
@@ -31,28 +32,46 @@ def load_plugins_from_dir(plugins_dir: str) -> List[PluginData]:
             plugin_data = load_plugin_from_zip(plugin_zip_path)
             plugins_data.append(plugin_data)
 
-    return sorted(plugins_data, key=lambda p: p.plugin_manifest.priority)
+    # sort by priority, then by version ascending
+    plugins_data.sort(key=lambda p: (p.plugin_manifest.priority, SemanticVersion(p.plugin_manifest.version)))
+    return plugins_data
 
 
 def load_plugin_from_zip(plugin_zip_path: Path) -> PluginData:
     assert plugin_zip_path.is_file(), f'no such file {plugin_zip_path}'
-    extracted_plugin_path = plugin_zip_path.parent / EXTRACTED_PLUGINS_DIR / plugin_zip_path.stem
     
+    extracted_plugins_dir = plugin_zip_path.parent / EXTRACTED_PLUGINS_DIR
+    if not extracted_plugins_dir.is_dir():
+        extracted_plugins_dir.mkdir(parents=True, exist_ok=True)
+        extracted_plugins_dir.chmod(mode=0o777)  # mkdir(mode=0o777) doesn't really work
```

Comment:
What it means doesnt' work? Leftover comment? Or if not, if 777 doesnt' work, should it be changed?

Reply:
I wrote a comment in case if anyone wonders why I didn't do it in one line: `extracted_plugins_dir.mkdir(parents=True, exist_ok=True, mode=0o777)`

Reply:
ah, so I got confused by comment, and thought it means `chmod(777)` doesn't work. I think the comment should be on mkdir line, or removed at all, as it's comment about python's file API.

Reply:
Alright, I'll get rid of it.